content/en/blog/_posts/2023-01-12-protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,289 @@
+---
+layout: blog
+title: "Protect Your Mission-Critical Pods From Eviction With PriorityClass"
+date: 2023-01-12
+slug: protect-mission-critical-pods-priorityclass
+description: "Pod priority and preemption help to make sure that mission-critical pods are up in the event of a resource crunch by deciding order of scheduling and eviction."
+---
+
+
+**Author:** Sunny Bhambhani (InfraCloud Technologies)
+
+Kubernetes has been widely adopted, and many organizations use it as their de-facto orchestration engine for running workloads that need to be created and deleted frequently.
+
+Therefore, proper scheduling of the pods is key to ensuring that application pods are up and running within the Kubernetes cluster without any issues. This article delves into the use cases around resource management by leveraging the [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) object to protect mission-critical or high-priority pods from getting evicted and making sure that the application pods are up, running, and serving traffic.
+
+## Resource management in Kubernetes 
+
+The control plane consists of multiple components, out of which the scheduler (usually the built-in [kube-scheduler](/docs/concepts/scheduling-eviction/kube-scheduler/)) is one of the components which is responsible for assigning a node to a pod.
+- Whenever a pod is created, it goes to a pending state, and then the scheduler kicks in and determines which node should be the best-suited node for the placement of the new pod.
+  - In the background, the scheduler runs an infinite loop looking for pods with null in the nodeName field.
+  - When it finds one, it starts determining which node should be the best one for the pod.
+  - There are various parameters for determining the pod's placement, but to name a few, they are `nodeName` , `nodeSelector` , `taints & tolerations` , `nodeAffinity` , rank of available nodes based on available resources in terms of CPU and memory, etc.
```

Comment:
Does this sound good? because the rank of nodes is also one of the parameters that is decided by the scheduler, so I wanted to have that as one of the values.

To name a few, "node selector," "taints and tolerances," "node affinity," the rank of nodes based on available resources (for example, CPU and memory," and several other criteria are used to determine the pod's placement. 

Reply:
Hi @sftim , 

Thank you for reviewing the article, I have made all the changes based on your recommendations.

Could you please have a quick check?

The preview version can be found here: https://deploy-preview-37713--kubernetes-io-main-staging.netlify.app/blog/2023/01/12/protect-mission-critical-pods-priorityclass/ 

PS: Also have corrected the typo in the image.

Thank you,
Sunny